pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -406,6 +409,12 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				}
 				gatewayRoutes[gatewayName][vskey] = routes
 			}
+			var gatewayService *model.Service
+			for _, hostname := range intersectingHosts {
+				if svc, exists := nameToServiceMap[hostname]; exists {
+					gatewayService = svc
```

Comment:
This could not get the gateway service when users are setting wildcard host "*"

I guess you could get the gateway service from node.ServiceInstances

Reply:
You mean wildcard host in VS? I think it will work. Have changed the test. Please check

Reply:
You are right. The gatewayService var actually means service that can be routed from gateway

Reply:
Can't `intersectingHosts` be something like `[foo.com, bar.com]`? Which are entirely unrelated?